src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -210,6 +228,70 @@ private boolean canGoPlanting(@NotNull Field currentField, @NotNull BuildingFarm
         return true;
     }
 
+    private AIState cycle()
+    {
+        @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+
+        if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
+        {
+            return AIState.PREPARING;
+        }
+
+        @Nullable Field field = buildingFarmer.getCurrentField();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)